### DISCUSSION TITLE
Replace remote control app on iPhone with Tasmota flashed bluetooth module HM-10

### DISCUSSION THREAD

**Discussion by francodutch** [arendst/Tasmota#11355] [Ideas]
I have an HM-10 bluetooth 4.0 module based on the CC2541 and I was wondering if I could somehow use it to replace the iphone remote control mobile app of my Daiko AC and that way control the AC via Home Assistant. I would need to somehow sniff the bluetooth signals sent from the app/iphone and send these via the Tasmota flashed HM-10. I have started reading up on bluetooth - has someone done this or is it not feasible?

**Comment by barbudor** [comment]
Hi
In Tasmota, HM-10 with ESP8266 is only used for receiving data from sensors and iBeacon. AFAIK, it is not possible to send data to a BT device.

On ESP32, the newly added Generic BLE driver may allow to send commands "BleOp Write" to devices but this is only in not connected mode.
I fear that those AirCon requires a connection to receive commands.

However this is probably easier to achieve on ESP32 by creating a driver on top of the generic BLE driver like it has been done for the EQ3 thermostats.
Let me call @btsimonh in the discussion who may be able to advise more.
